scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        --------
+        ContinuousDistribution.pdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support ``(l, r)``.
+        The following table summarizes the value returned by methods
+        of ``ContinuousDistribution`` for arguments outside the support.
+
+        +----------------+---------------------+---------------------+
+        | Method         | Value for ``x < l`` | Value for ``x > r`` |
+        +================+=====================+=====================+
+        | ``pdf(x)``     | 0                   | 0                   |
+        +----------------+---------------------+---------------------+
+        | ``logpdf(x)``  | -inf                | -inf                |
+        +----------------+---------------------+---------------------+
+        | ``cdf(x)``     | 0                   | 1                   |
+        +----------------+---------------------+---------------------+
+        | ``logcdf(x)``  | -inf                | 0                   |
+        +----------------+---------------------+---------------------+
+        | ``ccdf(x)``    | 1                   | 0                   |
+        +----------------+---------------------+---------------------+
+        | ``logccdf(x)`` | 0                   | -inf                |
+        +----------------+---------------------+---------------------+
+
+        For the ``cdf`` and related methods, the inequality need not be
+        strict; i.e. the tabulated value is returned when the method is
+        evaluated *at* the corresponding boundary.
+
+        The following table summarizes the value returned by the inverse
+        methods of ``ContinuousDistribution`` for arguments at the boundaries
+        of the domain ``0`` to ``1``.
+
+        +-------------+-----------+-----------+
+        | Method      | ``x = 0`` | ``x = 1`` |
+        +=============+===========+===========+
+        | ``icdf(x)`` | ``l``     | ``r``     |
+        +-------------+-----------+-----------+
+        | ``icdf(x)`` | ``r``     | ``l``     |
+        +-------------+-----------+-----------+
+
+        For the inverse log-functions, the same values are returned for
+        for ``x = log(0)`` and ``x = log(1)``. All inverse functions return
+        ``nan`` when evaluated at an argument outside the domain ``0`` to ``1``.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Retrieve the support of the distribution:
+
+        >>> X.support()
+        (-0.5, 0.5)
+
+        For a distribution with infinite support,
+
+        >>> X = stats.Normal()
+        >>> X.support()
+        (-inf, inf)
+
+        Due to underflow, the numerical value returned by the PDF may be zero
+        even for arguments within the support, even if the true value is
+        nonzero. In such cases, the log-PDF may be useful.
```

Comment:
Were do we land here for the RFC? Can we already give some precision guarantee and what is out of scope?

Otherwise +1, good addition to remind people about such numerical things.

Reply:
> Were do we land here for the RFC? Can we already give some precision guarantee and what is out of scope?

Just to be clear, accuracy guarantees on particular distributions are unrelated to this comment. This comment is about a fundamental limitation of floating point numbers, not the accuracy of an implementation.

> Can we already give some precision guarantee and what is out of scope?

Depends on what distributions we offer. I would expect for the standard normal and uniform distributions that we can be accurate on the order of machine precision for arbitrary parameters. But before we make a statement, we need that test suite which compares against mpmath, and that is not written yet.